### ISSUE TITLE
Logical Partition Deletion

### ISSUE THREAD

**Issue by KristianJakubik** [Azure/azure-cosmos-dotnet-v3#1672]
Hi,
I would like to ask question regarding logical partition deletion.
I've recently came across this [PR](https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1490) for **PKDeletePurger**, which has updated its implementation and adds new method `DeleteAllItemsByPartitionKeyAsync`. The name of PR says the implementation is meant for SDK 3.
**Can you please verify that this will be shipped with SDK 3 and if you have any estimate when?**
As a workaround I found that REST API of Cosmos Containers already implements ability to delete whole logical partition through calling this endpoint`[Container]/operations/partitionkeydelete`.
**I just want to verify that this approach is correct and production ready as I couldn't find much of an info about cosmos REST API changelog, is there any? 
And are there any gotchas when calling cosmos db directly via REST API ?**

**Comment by j82w** [comment]
@abhijitpai do you know the answer to this?

**Comment by abhijitpai** [comment]
Yes, it will be shipped on this SDK (aka v3). We expect it to be in a preview package in a couple of months by which time we will expose some metrics around the deletion, improve stability, add a sample and some documentation. There are a many caveats to using the API today and we will reduce some of them by preview - I would suggest to use alternatives in the mean time if they work for your use case: just query for and then delete items in parallel at the rate you want, or use the bulk support we have in the SDK to delete items with high throughput if you're okay with it using all of the provisioned container throughput, etc.

**Comment by KristianJakubik** [comment]
@abhijitpai thank you for explanation.

We did consider using query and bulk, but it didn't fit much our case. In our use case we don't need to have thorough diagnostics of the operation, handling non-successful response with exception or parsing additional response headers. Our key requirements are correctness of the operation and data consistency after the operation has finished. These requirements are sufficient for us so far, but I was afraid that those might have not been met yet. And I wanted to know if the API endpoint is in preview sort of mode or if it is publicly available on all SQL/Table API containers. 
It's quite hard to distinguish for us developers when the last update of rest api documentation (https://docs.microsoft.com/en-us/rest/api/cosmos-db/) was some time ago.

**Comment by j82w** [comment]
@deborahc for feedback about the REST API documentation.

**Comment by imshashwataggarwal** [comment]
@abhijitpai is this - https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1490 functionality out to consume? If yes can you share the .NET SDK version

**Comment by InquisitorJax** [comment]
@abhijitpai 👆?

**Comment by YohanSciubukgian** [comment]
Do you have any update to share on the release date of this feature ?

**Comment by abhijitpai** [comment]
This has been available in the preview SDK for a while now - please see the minimum version requirements and enroll in the preview as this page suggests: https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/how-to-delete-by-partition-key?tabs=dotnet-example#sdk-requirements

**Comment by InquisitorJax** [comment]
I've shipped app updates that use the preview version, and it's working as expected.
As mentioned though - it seems this has been stuck in preview status for a long time now - looking forward to not have to rely on preview builds to ship production code ;)

**Comment by olivermue** [comment]
So, we are in 2024 and the function is still in Preview only. Any chance or knowledge, when this function will make it into the official NuGet package?

**Comment by jcageman** [comment]
It seems to have been released in https://github.com/Azure/azure-cosmos-dotnet-v3/releases/tag/3.45.0

Could you also update the docs stating it's no longer in preview?
 https://learn.microsoft.com/en-us/azure/cosmos-db/nosql/how-to-delete-by-partition-key?tabs=dotnet-example
